Crear componente para renderizar sección "Nosotros" con información de la ONG

ArielSurco opened the pull request:
User story link: https://alkemy-labs.atlassian.net/browse/OT91-54

**##SUMMARY**
It show the top section of "About us" component in /nosotros route

**##CHANGES ADDED**
- Render AboutUs in /nosotros route
- Add AboutUsDescription component: Use Title component, displays "Nosotros" and receives a "description" property that is displayed below the title.
- Add AboutUs styles
- Use description const in AboutUs.js: Represents a description obtained with an API request and uses it to render the AboutUsDescription component

**##SCREENSHOTS**
![image](https://user-images.githubusercontent.com/50643955/139785450-e500eb82-d13a-4922-98e6-c23f904d1940.png)

Commit:
feat: :sparkles: add form create and edit projects

Commit:
config initial

Commit:
feat: :sparkles: agrega funciones de post, get, put para projects

Commit:
feat: :sparkles: agrega funciones de post, get, put para projects

Commit:
feat: :sparkles: add request http (get,put,post) for compoent form projects

Commit:
feat: :sparkles: add form for edit and create projects

Commit:
feat: :construction: agg request http (get,post,put) for form activity

Commit:
feat: :sparkles: add form for create and edit activities

Commit:
style: :lipstick: modify layout

Commit:
style: :lipstick: modify layout

Commit:
style: :lipstick: modification layout

MartinCaparros committed:
Component and routes created

MartinCaparros committed:
Organization data layout

Commit:
fix: :adhesive_bandage: modularization of the activity form

it extracts in two functions the rendering of errors, and the obtaining of the url of the image inout

Commit:
feat: :sparkles: add request HTTP ( get, post, put) for the form activity

Commit:
fix: :rewind: revert the changes of the project form

reverts the changes of the project form to its original state and deletes the folder created because it is from another unrelated activity

Commit:
feat: :sparkles: add modularized functions to get the url of an image

add modularized functions to get the url of an image, and save it in a state

Commit:
feat: :sparkles: adds modularized functions to render form errors

adds modularized functions to render form errors made with formik, requires name and error

Commit:
feat: :adhesive_bandage: modularization of the project form

it extracts in two functions the rendering of errors, and the obtaining of the url of the image inout

Commit:
feat: :sparkles: add request HTTP ( get, post, put) for the form project

Commit:
fix: :rewind: agg request http (get,post,put) for form project

Commit:
feat: :sparkles: add modularized functions to get the url of an image

add modularized functions to get the url of an image, and save it in a state

Commit:
feat: :sparkles: adds modularized functions to render form errors

adds modularized functions to render form errors made with formik, requires name and error

Commit:
feat: :sparkles: add form for edit and create testimonials

it extracts in two functions the rendering of errors, and the obtaining of the url of the image inout

Commit:
feat: :sparkles: add request HTTP ( get, post, put) for the form testimonial

Commit:
feat: :sparkles: add modularized functions to get the url of an image

add modularized functions to get the url of an image, and save it in a state

Commit:
feat: :sparkles: adds modularized functions to render form errors

adds modularized functions to render form errors made with formik, requires name and error

Commit:
fix: :adhesive_bandage: fix saved image data

fix the saved image data, now saving the 64bit url

Commit:
fix: :adhesive_bandage: fix saved image data

fix the saved image data, now saving the 64bit url

Commit:
delete console.log in disuse

Commit:
delete console.log in disuse

Commit:
fix: :adhesive_bandage: fix saved image data

fix the saved image data, now saving the 64bit url

Commit:
fix: :adhesive_bandage: fix saved image data

fix the saved image data, now saving the 64bit url

MartinCaparros committed:
Dismiss this changes

Commit:
feat: :sparkles: function to validate form the testimonials forms

function to validate form forms, for name, description and image fields, returning the corresponding errors

Commit:
feat: :sparkles: form modularization

removes the functionality of replace validate errors to other components in the common folder

Commit:
feat: :sparkles: create functions to validate formik form fields and return errors

It includes the replace function for the "<p></p>" of the ckditor component and a function to validate the values of the formik input

Commit:
feat: :sparkles: form modularization

extra the form and replace validations from ckeditor in a function in the common folder

Commit:
feat: :sparkles: create functions to validate formik form fields and return errors

It includes the replace function for the "<p></p>" of the ckditor component and a function to validate the values of the formik input

Commit:
feat: :sparkles: form modularization

extra the form and replace validations from ckeditor in a function in the common folder

MarCortina committed:
create services to endpoint

MarCortina committed:
add img category selected & onsubmit exec endpoint

MarCortina committed:
initial configuration eslint/pjson/prettier

MartinCaparros committed:
All changes done, service for org data retrieve created and working.

MartinCaparros committed:
I create a new route in App.js Create two component, one that recieves a tittle by prop and renders on a new component, also it render new data that comes from a callback to the endpoint with the ID from the new. I also create the service that makes the callback to the endpoint wich takes ID as parameter.

Commit:
feat: :fire: remove and delete de function in folder common

Commit:
feat: :sparkles: add function setCKEditorText

replace "<p></p>" in the text and set value in the state

Commit:
feat: :sparkles: adds validation folder, includes validations for the testimonial form and function to validate the length of the name or title

Commit:
feat: :sparkles: add function to render errors in the form

Commit:
feat: :sparkles: adds function to dispatch the corresponding request

Commit:
feat: :sparkles: modify the name of some variables and element

Commit:
feat: :fire: delete file

Commit:
feat: :seedling: modified route to import file

modified route to import file "setCKEditor" and "validations"

saralavanchy committed:
pull request #2 autenticacion_de_usuarios/formulario_login

Email and password fields validation with Formik.

saralavanchy committed:
Revert "Formulario Login"

saralavanchy committed:
pull request #18 from alkemyTech/revert-2-feat/autenticacion_de_usuarios/formulario_login

Revert "Formulario Login"

Commit:
feat: :fire: delete deprecated file

Commit:
feat: :sparkles: add file file with functions to read and set the value of input file

add the seturl function that invokes the function that reads the url and sets it through a callback function

Commit:
feat: :speech_balloon: modified name function "validateTestimonial => validateTestimonialForm"

Commit:
feat: :speech_balloon: modify the names of some functions

modify the names of some functions to make them more understandable and representative

Commit:
feat: :speech_balloon: modify the names of some functions to make them more understandable and representative

andreajhl committed:
Merge branch 'development' into feat/testimonial-form/create-and-edit-form-testimonials

Commit:
feat: :sparkles: modify variable names and separate functions in external folders

modify variable names to be more representative and extract functions in the common folder to make the component modularized and reusable

Commit:
feat: :fire: delete unused files

the functions in these files were moved to new files in their corresponding folders

Commit:
feat: :sparkles: add function to render form errors made with formik

Commit:
feat: :sparkles: add function to read and set url of input file

add function to read and set its value through a callback function

Commit:
feat: :sparkles: add function to remove the "<p> </p>" from the value of ckeditor

Commit:
feat: :sparkles: add function to validate the form fields and return the corresponding errors

Commit:
feat: :fire: delete unused files
the functions in these files were moved to new files in their corresponding folders

Commit:
feat: :sparkles: add function to render form errors made with formik

Commit:
feat: :sparkles: add function to read and set url of input file
add function to read and set its value through a callback function

Commit:
feat: :sparkles: add function to remove the "<p> </p>" from the value of ckeditor

Commit:
feat: :sparkles: add function to render form errors made with formik

Commit:
feat: :sparkles: modify variable names and separate functions in external folders
modify variable names to be more representative and extract functions in the common folder to make the component modularized and reusable

Commit:
feat: :fire: delete unused files
the functions in these files were moved to new files in their corresponding folders

Commit:
feat: :sparkles: modify variable names and separate functions in external folders
modify variable names to be more representative and extract functions in the common folder to make the component modularized and reusable

Commit:
feat: :sparkles: modify variable name

Commit:
feat: :sparkles: add function to dispatch the creation or modification request depending on whether there is or an id

Commit:
Merge branch development into feat/activity-Form/create-and-edit-form-activities

MartinCaparros committed:
I declare the function outside the useEffect()

MartinCaparros committed:
All spaces removed, function declaration outside of useEffect().

MarCortina committed:
add dropzone-react

MarCortina committed:
required changes category service

MarCortina committed:
change name required

MarCortina committed:
add input image component

MarCortina committed:
add input CKeditor component

MarCortina committed:
changes names & business logic required

Commit:
feat: :sparkles: add route for edit projects in file APP

Commit:
feat: :sparkles: modified name for validations form projects

modified "name => title"

Commit:
feat: :sparkles: modifies imports of functions and agg obtaining id by params

Commit:
feat: :sparkles: add new function to dispatch creation or modification and modified name for route

Commit:
feat: :sparkles: modified name route for import set UrlImage

andreajhl committed:
Merge branch 'development' into feat/project-Form/create-and-edit-form-projects

MarCortina committed:
add replace in data content

MarCortina committed:
deleted files

Commit:
feat: :sparkles: modified up_date a due_date

Commit:
Merge branch feat/project-Form/create-and-edit-form-projects of https://github.com/alkemyTech/OT91-server into feat/project-Form/create-and-edit-form-projects

Commit:
feat: :sparkles: modified route request api rest

Commit:
feat: :sparkles: modified route request api rest

ArielSurco committed:
Eslint corrections

ArielSurco committed:
Correct import error  of DefaultImage in Title.js

Commit:
feat: :sparkles: add route in APP for edit activity

Commit:
feat: :sparkles: modifies the initial value of the form momentarily, until you have redux

Commit:
feat: :sparkles: validation to decide which request to make

ArielSurco committed:
Render AboutUs in /nosotros route

Commit:
feat: :sparkles: validation to decide which request to make

Commit:
feat: :sparkles: add route in file App for render edit testimonial

Commit:
feat: :sparkles: validation to decide which request to make

Commit:
feat: :sparkles: modifies the initial value of the form momentarily, until you have redux

Commit:
feat: :fire: delete code in App

Commit:
feat: :fire: delete unused files

Commit:
feat: :fire: remove import to counter slider

Commit:
feat: :fire: remove import and delete file unused to counter

Commit:
feat: :fire: remove import and delete file unused to counter

ArielSurco committed:
Add AboutUsDescription component and AboutUs styles

ArielSurco committed:
Use AboutUsDescription component, with description const, in AboutUs

Commit:
Merge branch development into feat/project-Form/create-and-edit-form-projects

Commit:
Merge branch development into feat/activity-Form/create-and-edit-form-activities

andreajhl committed:
Merge branch 'development' into feat/testimonial-form/create-and-edit-form-testimonials

MartinCaparros committed:
Merge branch 'development' into feat/organizacion/screen_datos_de_organizacion_backoffice

MartinCaparros committed:
Add the " / " missing on App js Route.

MartinCaparros committed:
Merge branch 'development' into feat/novedades/crear_componente_detalle_novedad

saralavanchy committed:
Merge pull request #10 from alkemyTech/feat/organizacion/screen_datos_de_organizacion_backoffice

Screen de datos de organizacion (backoffice)

MarCortina committed:
Merge branch 'development' into feat/OT91/formulario-edicion-creacion

MarCortina committed:
Merge pull request #14 from alkemyTech/feat/OT91/formulario-edicion-creacion

Feat/News/formulario edicion creacion

saralavanchy committed:
Merge pull request #16 from alkemyTech/feat/novedades/crear_componente_detalle_novedad

Componente detalle de Novedad

andreajhl committed:
Merge branch 'development' into feat/testimonial-form/create-and-edit-form-testimonials

andreajhl committed:
Merge branch 'development' into feat/project-Form/create-and-edit-form-projects

andrespereyra-it committed:
Rename assets folder to Assets, delete duplicated default image

saralavanchy committed:
Merge pull request #26 from alkemyTech/bug/title/fix-assets-folder-name

Rename assets folder to Assets, delete duplicated default image

Commit:
Adding missing imports on App.js routes and solving lint errors

andreajhl committed:
Merge pull request #4 from alkemyTech/feat/project-Form/create-and-edit-form-projects

Feat/project form/create and edit form projects

andreajhl committed:
Merge branch 'development' into feat/activity-Form/create-and-edit-form-activities

andreajhl committed:
Merge branch 'development' into feat/testimonial-form/create-and-edit-form-testimonials

saralavanchy committed:
Merge branch 'development' into bug/general/fix-development-branch

saralavanchy commented on src/Components/Title/Title.js:
```diff
@@ -1,7 +1,7 @@
 
-import DefaultImage from "../../Assets/Title/default.jpg";
+import DefaultImage from "../../assets/Title/default.jpg";
 
```
this file is not scope of this PR

ArielSurco replied:
I made that change to fix the compile problem.
The other changes out of scope were eslint fixes, I have to solve it with a merge from development to this branch, right?

saralavanchy commented on src/Components/Newsletter/NewsletterForm.js:
```diff
@@ -5,7 +5,7 @@ const SetTokenNewsLetter = () =>{
     localStorage.setItem("tokenNL","settokenNL")
 }
```
this file is not scope of this PR

saralavanchy commented on src/Components/Users/UserModifierForm.js:
```diff
@@ -29,7 +29,7 @@ const UserModifierForm = ({user}) => {
             return true;
         };
```
this file is not scope of this PR

saralavanchy commented on src/Components/utilsData/roleList.js:
```diff
@@ -1 +1 @@
-export const rolesList = [{roleId:'Administrador'},{roleId:'Regular'}]
\ No newline at end of file
+export const rolesList = [{roleId:'Administrador'},{roleId:'Regular'}]
```
this file is not scope of this PR

saralavanchy commented on src/Services/userServices.js:
```diff
@@ -8,4 +8,4 @@ export const createUser = async (user) => {
 export const updateUser = async (id,user) => {
     const {data} = await axios.put(`http://ongapi.alkemy.org/api/users/${id}`,user)
     return data
-}
```
this file is not scope of this PR

saralavanchy commented on src/Styles/AboutUs.css:
```diff
@@ -0,0 +1,18 @@
+.about-us-description {
```
try to improve this file thinking on make this styles reusable

saralavanchy replied:
if this is a container, create a container file for styles and put this in it. Try to never use a particular stylesheet for a component.

saralavanchy commented on src/Components/Activities/ActivitiesForm.js:
```diff
@@ -15,7 +15,7 @@ const ActivitiesForm = () => {
       setInitialValues({ ...initialValues, description: e.target.value });
```
this file is not scope of this PR

saralavanchy commented on src/Components/About/AboutUsDescription.js:
```diff
@@ -0,0 +1,16 @@
+const AboutUsDescription = ({description}) => {
+    return (
+        <Box className="about-us-description">
+            <Title title="Nosotros"/>
```
the title should be on tje About Us Component and not on the description component

saralavanchy commented on src/Components/About/AboutUsDescription.js:
```diff
@@ -0,0 +1,16 @@
+    return (
+        <Box className="about-us-description">
+            <Title title="Nosotros"/>
+            <div className="about-us-description__text">
```
use the same logic for the names: about-us-description-text

saralavanchy committed:
Merge pull request #28 from alkemyTech/bug/general/fix-development-branch

[Bugfix] Adding missing imports and solving lint errors

andreajhl committed:
Merge branch 'development' into feat/activity-Form/create-and-edit-form-activities

saralavanchy committed:
Merge pull request #13 from alkemyTech/feat/testimonial-form/create-and-edit-form-testimonials

Feat/testimonial form/create and edit form testimonials

andreajhl committed:
Merge branch 'development' into feat/activity-Form/create-and-edit-form-activities

saralavanchy committed:
Merge pull request #5 from alkemyTech/feat/activity-Form/create-and-edit-form-activities

Feat/activity form/create and edit form activities

MartinCaparros committed:
Card created, its a component that recieves 3 props (title,img,description), in case that img its null it renders a placeholder.

Commit:
feat: :sparkles: add route "/activities" for render activities list

Commit:
feat: :sparkles: add component "title"and "Pagination"

Commit:
feat: :sparkles: step through the component array to pass the data to each card

Commit:
feat: :sparkles: contiene logica para agregar un paginado desde el front, asi se muestra solo 9 actividades por pantalla,

This sets the items to be displayed per page in a local state and sends them to activities cards for them to go through.

Commit:
feat: :sparkles: add components to render card, show image title and description

Commit:
feat: add function to remove the "<p> </p>" and "<strong> </strong>" by a space

Commit:
add request http get list activities

Commit:
feat: :sparkles: remove file

Commit:
feat: :bug: fix import files, and errors of spaces and empty lines

saralavanchy committed:
Merge pull request #34 from alkemyTech/bugfix/files/correccion-achivos

Bugfix/files/correccion achivos

andreajhl committed:
Merge branch 'development' into feat/component-activities/page-render-cards-activities

ArielSurco committed:
Add tokenExists function

ArielSurco committed:
Rename tokenExists to getAuthorizationHeader and fix it

saralavanchy committed:
Crear método que devuelva el encabezado de seguridad para ser reutilizado en las peticiones a los metodos privados de la API

Crear método que devuelva el encabezado de seguridad para ser reutilizado en las peticiones a los metodos privados de la API

MartinCaparros committed:
Logic updated.

I change the logic in the lines 24 and 27.

MartinCaparros committed:
Home folder, index.js, route created.

MarCortina committed:
add edit or create member component

MarCortina committed:
add edit member component & dlt duplicates imports

saralavanchy committed:
Adding CustomCard component

CustomCard component

Commit:
feat: :fire: remove deprecated code

Commit:
add text const folder

Commit:
feat: :sparkles: add component to render cards

use the ckeditor function to remove the html code from the description

Commit:
feat: :sparkles: add reusable component to create pagination from front end

receives a function that renders the list and the items in an array that will divide

Commit:
add text const folder

Commit:
modify the above function to also remove "<strong> </strong>"

Commit:
feat: :sparkles: add component to render cards

use the ckeditor function to remove the html code from the description and use constants for texts, so they can be changed more easily in the future

Commit:
add component that renders the pagination with the cards and the title of the page

julian624 committed:
Created slides services base http!

MartinCaparros committed:
Merge branch 'development' into feat/home/crear_componente_y_estructura_para_renderizar_pagina_inicio

saralavanchy committed:
Home layout created

Home layout created.

julian624 committed:
just deleted blank spaces

julian624 committed:
Created ActivitiesContent component

julian624 committed:
ADDED REACT HTML PARSER AND FIXED JSON SPACES

julian624 committed:
Commit to try to fix

MarCortina committed:
add validation to image, name, social media

MarCortina committed:
add import validation, handler, alerts

MarCortina committed:
add alerts error & success

MarCortina committed:
add handlers ti CKEditor form

Commit:
applying yarn lint to solve lint issues

saralavanchy committed:
[BugFix] Solving lint issues on development

[BugFix] Solving lint issues on development

andreajhl committed:
Merge branch 'development' into feat/component-activities/page-render-cards-activities

TelmaKn committed:
Recreation of all files

saralavanchy committed:
Crear componente para renderizar sección "Actividades"

saralavanchy committed:
Extender el servicio HTTP base para las peticiones de Slides

Feat/slides/extender el servicio http base

saralavanchy committed:
Crear componente contenido de actividad

Feat/actividades/crear componente de actividades

MarCortina committed:
resolve conflicts in app

MarCortina committed:
resolve conflicts in app

MarCortina committed:
resolve conflicts in app

MarCortina committed:
resolve conflicts in app

MarCortina committed:
resolve conflicts in app

MarCortina committed:
resolve conflicts in app

saralavanchy committed:
Formulario Edición / Creación de Miembros

Member edit or create

saralavanchy committed:
Formulario Edición / Creación de Categorías

Formulario Edición / Creación de Categorías

ArielSurco committed:
Render AboutUs in /nosotros route

ArielSurco committed:
Add AboutUsDescription component and AboutUs styles

ArielSurco committed:
Use AboutUsDescription component, with description const, in AboutUs

ArielSurco committed:
Merge branch 'feat/nosotros/crear_componente_para_renderizar_seccion_nosotros' of github.com:alkemyTech/OT91-server into feat/nosotros/crear_componente_para_renderizar_seccion_nosotros

ArielSurco committed:
Delete unnnecessary folder

ArielSurco committed:
Rename AboutUs styles to make them reusable

ArielSurco committed:
Remove AboutUsDescription component and add it in AboutUs
